scipy/interpolate/interpnd.pyx

```diff
@@ -138,21 +191,24 @@ class NDInterpolatorBase:
             Points where to interpolate data at.
             x1, x2, ... xn can be array-like of float with broadcastable shape.
             or x1 can be array-like of float with shape ``(..., ndim)``
-
+        values : ndarray, optional
+            ndarray of float or complex, shape (npoints, ...)
+            Data values.
         """
-        xi = _ndim_coords_from_arrays(args, ndim=self.points.shape[1])
-        xi = self._check_call_shape(xi)
-        shape = xi.shape
-        xi = xi.reshape(-1, shape[-1])
-        xi = np.ascontiguousarray(xi, dtype=np.double)
+        if len(args) > 0:
+            self._set_xi(*args)
+        elif self._xi is None:
+            raise ValueError("Interpolation was called without required points")
+        if self.values is None:
+            raise ValueError("Values at interpolation points were not set")
```

Comment:
Actually the intention was to avoid the operation of finding the barycentric coordinates of points from `xi` in the triangulation.
In order to do so, the cython `evaluate` function should not perform this operation if it was already done, and I am not quite sure how to enable this behaviour for subclasses which won't need to reimplement `do_evaluate`, without saving some kind of state.
What do you suggest?

Reply:
My suggestion is to wrap a call to `_find_simplicies` (or its analog), `_set_xi` and other things into calls that return the necessary info. Then the `CloughTocherInterpolator` class does not store anything on an instance. Your subclass 
- overrides `__init__` to store stuff you want to store on the instance,
- overrides `_find_simplices` et all to look up instance variables.

This way the library version remains simple, and your subclass you can customize however you see fit.

Reply:
I have changed the functions accordingly.
Now the original signatures remain, and overriding `_find_simplices` should enable my requested fuctionality.